### PR TITLE
Fix several issues noted in #343 and #347

### PR DESCRIPTION
**Pull request by henry2cox**

### CODE REVIEW

**Comment by hartwork** [comment]
@henry2cox if #347 is fixed here, we can drop line…

https://github.com/linux-test-project/lcov/blob/e4b3129745f7e1434f26697bbc4c7027b96dd070/.github/workflows/run_test_suite.yml#L68

…here and have the CI confirm (to some extent) that the fix works as expected.

**Comment by henry2cox** [comment]
> @henry2cox if #347 is fixed here, we can drop line…

ah..right.  Oversight.
Will update the version in my sandbox and force-push again (I assume that that is an acceptable use model?  Or do we expect something different?)

**Comment by hartwork** [comment]
> Will update the version in my sandbox and force-push again (I assume that that is an acceptable use model? Or do we expect something different?)

@henry2cox I believe there are two conflicting schools about it: school A (including me) would say the branch is yours and yours only you can force push into at until it until its merged and school B (including CPython) would ask to only add new commits on top for all changes you make during review.

File: .github/workflows/run_test_suite.yml [hunk 1]
```diff
@@ -65,7 +65,6 @@ jobs:
 
           sudo perl -MCPAN -e 'install(Memory::Process)'  # no package in Ubuntu
```
Comment (hartwork) [inline .github/workflows/run_test_suite.yml:67]: (just an idea: I think that blank line before could also be dropped)

**Comment by hartwork** [comment]
@henry2cox the CI likes it :tada:

File: .github/workflows/run_test_suite.yml (hunk 1 above)
Comment (henry2cox) [reply .github/workflows/run_test_suite.yml:66]: sure...why not?

File: .github/workflows/run_test_suite.yml [hunk 2]
```diff
@@ -62,10 +62,8 @@ jobs:
 
           sudo apt-get update
           sudo apt-get install --no-install-recommends --yes -V "${ubuntu_packages[@]}"
-
```
[the remark this answers is not in the document]
Comment (henry2cox) [reply .github/workflows/run_test_suite.yml:65]: sigh.

[the remark this answers is not in the document]
Comment (hartwork) [reply .github/workflows/run_test_suite.yml:65]: @henry2cox looks just right now though :+1:

**Comment by hartwork** [comment]
@henry2cox my vote for merging but without squash-merge to keep the commit cutting intact on `master`.

**Pull request merged**